lib/carrierwave/storage/aws_file.rb

```diff
@@ -19,6 +19,12 @@ def file
         @file ||= bucket.object(path)
       end
 
+      def size
+        file.size
+      rescue Aws::S3::Errors::NotFound
+        nil
```

Comment:
Not sure about other adapters, but it is certainly fair to say the value would be undetermined.